src/main/java/uk/gov/companieshouse/authcodenotification/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.companieshouse.authcodenotification.exception;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.owasp.encoder.Encode;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import uk.gov.companieshouse.authcodenotification.utils.ApiLogger;
+
+import java.util.HashMap;
+
+import static uk.gov.companieshouse.authcodenotification.utils.Constants.ERIC_REQUEST_ID_KEY;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+    @Value("${GLOBAL_EXCEPTION_HANDLER_TRUNCATE_LENGTH_CHARS:15000}")
+    private int truncationLength;
+
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<Object> handleException(Exception ex, WebRequest webRequest) {
+
+        var context = webRequest.getHeader(ERIC_REQUEST_ID_KEY);
+        var sanitisedExceptionMessage = truncate(Encode.forJava(ex.getMessage()));
```

Comment:
If we sanitise for exceptions here then it might be good to rename the DataSanitiser added recently to LoggingDataSanitiser.